typescript/sdk/src/providers/MultiProvider.ts

```diff
@@ -550,7 +550,14 @@ export class MultiProvider {
     tx: PopulatedTransaction,
     from?: string,
   ): Promise<BigNumber> {
-    const txReq = await this.prepareTx(chainNameOrId, tx, from);
+    const txReq = {
+      ...(await this.prepareTx(chainNameOrId, tx, from)),
+      // Reset any tx request params that may have an unintended effect on gas estimation
+      gasLimit: undefined,
```

Comment:
from the PR description:
```
if some tx params like maxPriorityFeePerGas or maxFeePerGas were specified when we estimate gas, it can fail if the balance of the sender is insufficient. For cases where we want to estimate gas from the owner multisig which may not have native tokens, this caused issues.
```

The specific thing I ran into was for Ethereum both those `maxPriorityFeePerGas` and `maxFeePerGas` were specified when calling eth_estimateGas (bc they're present in the ethereum tx override), and bc the sender was the SAFE which only has like 0.05 ETH, the RPC would fail. It'd cap the gas limit as whatever we could afford, which was about 66k gas